protobuf-format.md

```diff
@@ -0,0 +1,387 @@
+to be represented in the Protocol buffers version 3.
+
+Protocol buffers are a language-neutral, platform-neutral extensible
+mechanism for serializing structured data. A message is defined once
+using the protobuf interface description language (IDL), and the
+protobuf compiler generates the language specific libraries for
+serializing to and deserializing from the binary representation. The
+protobuf version 3 library can also convert a protobuf message into
+its [standard JSON form][PROTO_JSON].
+
+Some built in message types in protobuf 3 (`google.protobuf.Value`,
+`google.protobuf.Any`, etc) have special runtime support from the
+protobuf library. These built in messages are given special treatment
+when transcoding to and from JSON. Tools in the protobuf ecosystem
+also give special treatment to these messages. For example, the
+[Common Expression Language tool (CEL)][CEL] automatically converts a
+`google.protobuf.Any` into its contained message when performing data
+queries.
+
+
+### 1.1. Conformance
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT",
+"SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this
+document are to be interpreted as described in RFC2119.
+
+## 2. Protobuf format
+
+Users of protobuf MUST use the official message definition when
+expressing a CloudEvent as a protobuf message. Users SHOULD NOT change
+the language specific options in order to have consistent code
+generation output. Users MUST NOT change the protobuf package
+name. Users MUST NOT use any protobuf options that change the protobuf
+wire compatibility such as message_set_wire_format.
+
+```proto
+syntax = "proto3";
+
+package io.cloudevents.v0;
+
+import "google/protobuf/any.proto";
+import "google/protobuf/struct.proto";
+import "google/protobuf/timestamp.proto";
+
+option go_package = "cloudevents.io/protobuf/";
+option java_package = "io.cloudevents";
+option java_multiple_files = true;
+
+message CloudEvent {
+  string event_type = 1;
+  string event_type_version = 2;
+  string cloud_events_version = 3;
+  string source = 4;
+  string event_id = 5;
+  google.protobuf.Timestamp timestamp = 6;
+  string schema_url = 7;
+  string content_type = 8;
+  // `oneof payload` itself is not a real field, it only enforces the
+  // `oneof` constraint
+  oneof payload {
+    google.protobuf.Value data = 9;
+    bytes bytes_data = 10;
+    google.protobuf.Any proto_data = 11;
+  }
+  google.protobuf.Struct extensions = 12;
+}
+```
+
+In general, the CloudEvents attribute names are converted into
+protobuf fields in lower_snake_case, and the the CloudEvents type
+system is mapped into protobuf types as follows:
+
+
+| CloudEvents | Protobuf
+|--------------|-------------------------------------------------------------
+| String       | string
+| Binary       | bytes
+| URI          | string
+| Timestamp    | google.protobuf.Timestamp
+| Map          | google.protobuf.Struct
+| Object       | google.protobuf.Value
+| Integer | int32
```

Comment:
minor thing, but if you're going to go thru the trouble to align most of the columns, you might as well align all of them  :-)